src/state/ducks/todos/todoSlice.ts

```diff
@@ -0,0 +1,179 @@
+import { createSlice, PayloadAction } from "@reduxjs/toolkit";
+import { ITodosState } from "state/ducks/todos/types/redux";
+import { ITodo } from "state/ducks/todos/types/utils";
+
+const initialState: ITodosState = {
+  data: [],
+  loading: false,
+  error: null,
+  alert: {
+    isOpen: false,
+    message: null,
+    severity: "success",
+  },
+};
+
+const todoSlice = createSlice({
+  name: "todos",
+  initialState,
+  reducers: {
+    getTodosRequest: (state) => {
+      return {
+        ...state,
+        loading: true,
+        error: null,
+      };
+    },
+    getTodosSuccess: (state, action: PayloadAction<ITodo[]>) => {
+      return {
+        ...state,
+        data: action.payload,
+        loading: false,
+        error: null,
+        alert: {
+          ...state.alert,
+          isOpen: false,
+        },
+      };
```

Comment:
why do we need this state? remove it as alerts can be handled directly from components.